lib/app-common.ts

```diff
@@ -156,33 +171,19 @@ async function setupGlobalAgentProxy(settings: CliSettings) {
 
 	const env = process.env;
 	env.GLOBAL_AGENT_ENVIRONMENT_VARIABLE_NAMESPACE = '';
+	env.NO_PROXY = [
+		...requiredNoProxy,
+		...(noProxy ? noProxy.split(',').filter(v => v) : privateNoProxy),
+	].join(',');
```

Comment:
I've taken the liberty to add my own commit that moves the 4 lines above from the `if (proxy)` block, so that BALENARC_NO_PROXY is also considered when BALENARC_PROXY is not set (but HTTP(S)_PROXY are set).